### PR TITLE
Update config version to 3

### PR DESCRIPTION
**Pull request by LexLuthr**
1. Remove redundant config from boost deal making
2. Update config version to 3
3. Add "HTTPRetrievalMultiaddr" to the file by default

### CODE REVIEW

File: node/config/migrate.go [hunk 1]
```diff
@@ -15,13 +15,14 @@ var log = logging.Logger("cfg")
...
 // We need to migrate the config file to this version.
-const CurrentVersion = 2
+const CurrentVersion = 3
 
 type migrateUpFn = func(cfgPath string) (string, error)
 
 var migrations = []migrateUpFn{
 	v0Tov1, // index 0 => version 0
-	v1Tov2, // index 1 => version 1
+	v1Tov2, // index 1 => version 3
```
Comment (dirkmc) [inline node/config/migrate.go:24]:
```suggestion
	v1Tov2, // index 1 => version 2
```

File: node/config/v2_to_v3.go [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+		return "", fmt.Errorf("parsing config file %s: %w", cfgPath, err)
+	}
+
+	boostCfg, ok := cfg.(*Boost)
+	if !ok {
+		return "", fmt.Errorf("unexpected config type %T: expected *config.Boost", cfg)
+	}
+
+	// For the migration from v1 to v2 just remove fields and update config version and add
+	// comments to the file
```
Comment (dirkmc) [inline node/config/v2_to_v3.go:20]: I don't think we need this comment

**Review by dirkmc — COMMENTED** [review]
A couple of nits but looks good 👍 

Are there any other boost deal-making params that are unused?

**Comment by LexLuthr** [comment]
I don't see any other unused config.

File: node/config/v2_to_v3.go (hunk 2 above)
Comment (dirkmc) [reply node/config/v2_to_v3.go:19]: I think you may have removed the wrong comment 😆

Comment (LexLuthr) [reply node/config/v2_to_v3.go]: Fixed.

**Review by dirkmc — APPROVED** [review]

**Pull request merged**